lib/hexpm/web/controllers/dashboard_controller.ex

```diff
@@ -28,6 +28,50 @@ defmodule Hexpm.Web.DashboardController do
     end
   end
 
+  def keys(conn, _params) do
+    user = conn.assigns.current_user
+    keys = Keys.all(user)
+
+    render_keys(conn, keys)
+  end
+
+  def revoke_key(conn, %{"name" => name} = params) do
+    user = conn.assigns.current_user
+    key = Hexpm.Repo.one(Key.get(user, params["name"]))
+
+    if key do
+      case Hexpm.Repo.update(Key.revoke(key)) do
```

Comment:
please use `Keys.remove(key, params["name"], audit: audit_data(conn)`

Reply:
Can you please throw some light on what purpose does `audit_data(conn)` serve?

Reply:
we keep an audit log of actions that change data (adding key, publishing package, removing owners etc), audit_data is used to get current user and their user agent.